controllers/release/release_adapter_test.go

```diff
@@ -0,0 +1,421 @@
+		}, time.Second*10).Should(BeTrue())
+
+		pipelineRun, _ := adapter.getReleasePipelineRun()
+		Expect(reflect.TypeOf(pipelineRun)).
+			To(Equal(reflect.TypeOf(&tektonv1beta1.PipelineRun{})))
+		Expect(pipelineRun).NotTo(BeNil())
+
+		// clean up
+		Expect(k8sClient.Delete(ctx, pipelineRun)).Should(Succeed())
+	})
+
+	It("ensures the Release Pipeline is tracked", func() {
+		Eventually(func() bool {
+			result, err := adapter.EnsureReleasePipelineRunExists()
+			return (!result.CancelRequest && err == nil)
+		}, time.Second*10).Should(BeTrue())
+		pipelineRun, _ := adapter.getReleasePipelineRun()
+		Expect(pipelineRun).NotTo(BeNil())
+
+		result, err := adapter.EnsureReleasePipelineStatusIsTracked()
+		Expect(!result.CancelRequest && err == nil).To(BeTrue())
+
+		// should return results.ContinueProcessing() in case the release is done
+		release.MarkSucceeded()
+		result, err = adapter.EnsureReleasePipelineStatusIsTracked()
+		Expect(!result.CancelRequest && err == nil).To(BeTrue())
+
+		// release is running but has no pipelineRun matching the release
+		release.MarkRunning()
+		pipelineRun.ObjectMeta.Labels = nil
+		k8sClient.Update(ctx, pipelineRun)
+		release.Status.Conditions = []metav1.Condition{}
+		result, err = adapter.EnsureReleasePipelineStatusIsTracked()
+
+		// err should not be nil here but adapter.getReleasePipelineRun does not advise
+		// there is no pipelineRun for the release
+		Expect(!result.CancelRequest && err == nil).To(BeTrue())
+
+		// local cleanup
+		k8sClient.Delete(ctx, pipelineRun)
+	})
+
+	It("ensures Finalizers are called for a Release object", func() {
+		// Call to EnsureFinalizersAreCalled should succeed if the Release was not set to be deleted
+		result, err := adapter.EnsureFinalizersAreCalled()
+		Expect(result.CancelRequest).To(BeFalse())
+		Expect(err).NotTo(HaveOccurred())
+
+		// Add finalizer
+		result, err = adapter.EnsureFinalizerIsAdded()
+		Expect(result.CancelRequest).To(BeFalse())
+		Expect(err).NotTo(HaveOccurred())
+
+		// Create a ReleasePipelineRun to ensure it gets finalized
+		result, err = adapter.EnsureReleasePipelineRunExists()
+		Expect(result.CancelRequest).To(BeFalse())
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(k8sClient.Delete(ctx, release)).NotTo(HaveOccurred())
+		Eventually(func() bool {
+			err = k8sClient.Get(ctx, types.NamespacedName{
+				Name:      release.Name,
+				Namespace: release.Namespace,
+			}, release)
+			return err == nil && release.GetDeletionTimestamp() != nil
+		}, time.Second*10).Should(BeTrue())
+
+		result, err = adapter.EnsureFinalizersAreCalled()
+		Expect(result.RequeueRequest).To(BeTrue())
+		Expect(err).NotTo(HaveOccurred())
+
+		Eventually(func() error {
+			return k8sClient.Get(ctx, types.NamespacedName{
+				Name:      release.Name,
+				Namespace: release.Namespace,
+			}, release)
+		}, time.Second*10).Should(HaveOccurred())
+
+		Eventually(func() bool {
+			pipelineRun, _ := adapter.getReleasePipelineRun()
+			return (pipelineRun == nil)
```

Comment:
Redundant parenthesis:
```suggestion
			return pipelineRun == nil
```

Reply:
fixed